server/src/printers/NativeMakefilePrinter.cpp

```diff
@@ -563,4 +668,66 @@ namespace printer {
                                                          bool exeToLib) {
         addLinkTargetRecursively(unitFile, suffixForParentOfStubs, false, exeToLib);
     }
+
+    void NativeMakefilePrinter::tryChangeToRelativePath(std::string& argument) const {
+        if (StringUtils::startsWith(argument, "/")) {
+            argument = getRelativePath(argument);
+            return;
+        }
+        // if in -I flag
+        if (argument.length() >= 3 && StringUtils::startsWith(argument, "-I")) {
+            argument = "-I" +
+                    getRelativePath(argument.substr(2)).string();
+        }
+    }
+
+    fs::path NativeMakefilePrinter::getRelativePath(const fs::path& source) const {
+        std::vector<std::string> basePaths = CollectionUtils::transformTo<std::vector<std::string>>(pathToShellVariable,
+                [](auto const& item) {
+            return item.first;
+        });
+        std::sort(basePaths.begin(), basePaths.end(), [](const std::string& first, const std::string& second) {
+            if (second == Paths::getUTBotRootDir().string()) {
+                return true;
+            }
+            return std::greater<>()(first, second);
+        });
+
+        for (const auto& path : basePaths) {
+            std::optional<fs::path> relative = Paths::getRelativePathWithShellVariable(
+                    pathToShellVariable.at(path),
+                    path,
+                    source);
+            if (relative.has_value()) {
+                return relative.value();
+            }
+        }
+        // don't change path if can't convert to relative
+        return source;
+    }
+
+    void NativeMakefilePrinter::declareShellVariable(const std::string& variableName, fs::path path,
+                              void (NativeMakefilePrinter::*declareFunc) (const std::string&, const std::string&),
+                              bool shouldWriteToMap) {
+        const fs::path relativePath = getRelativePath(path);
+        (*this.*declareFunc)(variableName, relativePath.string());
+        if (shouldWriteToMap) {
+            pathToShellVariable[path] = StringUtils::stringFormat("$(%s)", variableName);
+        }
+    }
+
+    fs::path NativeMakefilePrinter::getLinker(const fs::path& pathToLinker) const {
+        if (StringUtils::contains(pathToLinker.filename().c_str(), "ar")) {
```

Comment:
try readsymlink

Reply:
added parse for compiler/linker otherwise resolve by canonical